src/transferslot.cpp

```diff
@@ -433,6 +433,20 @@ void TransferSlot::doio(MegaClient* client)
                 continue;
             }
 
+            if (reqs[i]->status == REQ_FAILURE && reqs[i]->httpstatus == 200 && transfer->type == GET && transferbuf.isRaid())
+            {
+                // check if we got some data and the failure occured partway through the part chunk.  If so, best not to waste it, convert to success case with less data
+                HttpReqDL *downloadRequest = (HttpReqDL *)reqs[i];
+                LOG_debug << "Connection " << i << " received " << downloadRequest->bufpos << " before failing, processing data.";
+                if (downloadRequest->contentlength == downloadRequest->size && downloadRequest->bufpos >= RAIDSECTOR)
+                {
+                    downloadRequest->bufpos -= downloadRequest->bufpos % RAIDSECTOR;  // always on a raidline boundary
+                    downloadRequest->size = unsigned(downloadRequest->bufpos);
```

Comment:
`static_cast`?

Reply:
I think this is an old update to suppress warnings.  I prefer the shorter syntax for these, or changing the variable types to match but I don't think we can change the type of either of those.   Since static_cast was deliberately designed to be verbose and ugly to discourage its use, I'd rather use the shorter expression here?